app/grants/views.py

```diff
@@ -0,0 +1,430 @@
+        'subscription': {},
+        'grant_has_no_token': True if grant.token_address == '0x0000000000000000000000000000000000000000' else False,
+        'grant': grant,
+        'keywords': get_keywords(),
+    }
+    return TemplateResponse(request, 'grants/fund.html', params)
+
+
+@login_required
+def subscription_cancel(request, grant_id, grant_slug, subscription_id):
+    """Handle the cancellation of a grant subscription."""
+    subscription = Subscription.objects.select_related('grant').get(pk=subscription_id)
+    grant = getattr(subscription, 'grant', None)
+    now = datetime.datetime.now()
+    profile = request.user.profile if request.user.is_authenticated else None
+
+    if not subscription.active:
+        params = {
+            'active': 'grant_error',
+            'title': _('Grant Subscription Cancelled'),
+            'grant': grant
+        }
+
+        if grant.active:
+            params['text'] = _('This Grant subscription has already been cancelled.')
+        else:
+            params['text'] = _('This Subscription is already cancelled as the grant is not longer active.')
+
+        return TemplateResponse(request, 'grants/shared/error.html', params)
+
+    if request.method == 'POST' and (
+        profile == subscription.contributor_profile or request.user.has_perm('grants.change_subscription')
+    ):
+        subscription.active = False
+        subscription.save()
+        support_cancellation(grant, subscription)
+        messages.info(request, _('Your subscription has been canceled. We hope you continue to support other open source projects!'))
+        return redirect(reverse('grants:details', args=(grant.pk, grant.slug)))
+
+    params = {
+        'active': 'cancel_grant',
+        'title': _('Cancel Grant Subscription'),
+        'card_desc': _('Provide sustainable funding for Open Source with Gitcoin Grants'),
+        'subscription': subscription,
+        'grant': grant,
+        'now': now,
+        'keywords': get_keywords(),
+    }
+
+    return TemplateResponse(request, 'grants/cancel.html', params)
+
+
+@login_required
+def profile(request):
+    """Show grants profile of logged in user."""
+    if not request.user.has_perm('grants.view_grant'):
+        messages.info(request, _('You do not have permission to view grants.'))
+        return redirect(reverse('grants:grants'))
+
+    limit = request.GET.get('limit', 25)
+    page = request.GET.get('page', 1)
+    sort = request.GET.get('sort', '-created_on')
+
+    profile = request.user.profile if request.user.is_authenticated and request.user.profile else None
+    _grants = Grant.objects.prefetch_related('team_members') \
+        .filter(Q(admin_profile=profile) | Q(team_members__in=[profile])).order_by(sort)
+    sub_grants = Grant.objects.filter(subscriptions__contributor_profile=profile).order_by(sort)
+
+    paginator = Paginator(_grants, limit)
+    grants = paginator.get_page(page)
+
+    history = [{
```

Comment:
this is hard coded / needs to be driven by the backend

Reply:
Waiting on the miner